runtime/autoload/remote/host.vim

```diff
@@ -177,19 +178,24 @@ endfunction
 
 " Registration of standard hosts
 
-" Python {{{
+" Python/Python3 {{{
 function! s:RequirePythonHost(name)
+  let ver = (a:name ==# 'python') ? 2 : 3
```

Comment:
I've noticed that this will cause `ver` to be 3 always when called with `'legacy-python-provider'`.

This gets registered via `RegisterClone('legacy-python-provider', 'python')`. Maybe that should store a `orig_name` and use it here?